migration/idempotent/001-base.sql

```diff
@@ -1470,7 +1470,7 @@ CREATE OR REPLACE FUNCTION _prom_catalog.get_metric_retention_period(schema_name
 AS $$
     SELECT COALESCE(m.retention_period, _prom_catalog.get_default_retention_period())
     FROM _prom_catalog.metric m
-    WHERE id IN (SELECT id FROM _prom_catalog.get_metric_table_name_if_exists(schema_name, get_metric_retention_period.metric_name))
+    WHERE id IN (SELECT id FROM _prom_catalog.get_metric_table_name_if_exists(get_metric_retention_period.schema_name, get_metric_retention_period.metric_name))
```

Comment:
Usually `in` can always be rewritten as `exists` and it will be at least as slow, but might be much, much faster. Given it is a function, there is no way for pg to early interrupt the call, but I think it still might be worth it to rewrite here.